scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -192,6 +193,8 @@ sealed trait Execution[+T] extends java.io.Serializable {
    */
   def zip[U](that: Execution[U]): Execution[(T, U)] =
     Zipped(this, that)
+
+  override val hashCode = super.hashCode
```

Comment:
Looking into this further, there are actually a few other obstacles to having this work.  For example, even `Execution.from(5)` is going to create a closure object, and multiple calls won't make things that are "equal"; on develop:

```
scala> Execution.from(5) == Execution.from(5)
res0: Boolean = false

scala> Execution.from(5).hashCode
res1: Int = 765470035

scala> Execution.from(5).hashCode
res2: Int = -1375044982
```

I wouldn't mind fixing up that sort of thing.  From what I can tell it would involve making a few more case classes that wrap the functions we use here.  For example:

```
  private case class FutureTryWrapper[T](t: Try[T]) extends Function1[ConcurrentExecutionContext, Future[T]] {
    def apply(context: ConcurrentExecutionContext) = toFuture(t)
  }
  def fromTry[T](t: => Try[T]): Execution[T] = fromFuture(FutureTryWrapper(t))
```

Alternately I can just make sure we preserve the consistency we have today.

On a side note, any reason not to have `ReaderExecution`-based executions be lazy vals?